Revenue bonuses

dan-snw opened the pull request:
Fixes #1
Fixes #2
Fixes #3.

## Changes

I grouped together the revenue bonus buttons at the bottom, including the existing 'house has garden' button and a new CFO button. Grouping these at the end makes the logic clearer to the user (bonuses are added at the end and do not affect unit price so should be displayed at the end) - this also makes more space so the waitress functionality can be added later.

I made the house has garden button is smaller b adjusting the text to  just "Garden", which matches the new button "CFO", solving issue #2. At a later date I think it would be prettier to use an image here, but that should be done alongside other items too to avoid confusion since all of them are currently labelled by text. I can open a separate issue on this if this PR is accepted.

The CFO button uses the same logic as the garden button, though the price logic is a 1.5 multiplier to the total cash revenue - this is the method preferred in the original issue #3 description so that waitresses functionality can be added easily at a later point.

I also updated the ToggleButton to take an optional color argument, and updated the CFO and '-$1 Sale!' buttons to be purple and pink (solving issue #1) respectively. These colors are not hard coded when the components are created, but are given by an additional object ColorToggleButton which maps color names like 'pink' to rgb values so as to make it easier to give separate buttons the same color and later change the rgb of each subtly.

Before the changes:

![before](https://github.com/egnwd/food-chain-magnate/assets/19627954/56bfd2bf-c8af-46db-8d93-325933109a06)

After the changes:


![after](https://github.com/egnwd/food-chain-magnate/assets/19627954/a08c41b6-d07b-4bcc-981e-f3bb7e3e2618)

dan-snw committed:
Fixes issue #1: '-1$ Sale!' button now pale pink

Added optional parameter to the toggle button component - boolean to say
whether component is a permanent sale button or not. Then fixed the
toggle button component to be pink if it's a permanent sale button and
blue otherwise.

dan-snw committed:
Adds CEO button and makes group of revenue bonuses

Changed the order so that the garden button goes with the cfo bonus
since these both apply to the revenue but not the unit price. I put them
below the demand part too since they are "add-ons" to the demand
calculatoin, again making the logic of calculation slightly more
implicitly clear to the user.

dan-snw committed:
Refactors to pass toggle button colors

The ToggleButton component now includes an optional string argument for
the color instead of the flag for whether it should be pink. The default
is set as the existing default blue.

I also created an object to provide rgb colors by name that can be
imported when colors are needed, eg. ToggleButtonColors.purple.

The CFO button is now purple.

egnwd commented on src/Components/Price.tsx:
```diff
@@ -1,15 +1,16 @@
     let burgerBonus = demand.burgers * (milestones.hasBurgerBonus ? 5 : 0);
     let pizzaBonus = demand.pizza * (milestones.hasPizzaBonus ? 5 : 0);
-    return housePrice + drinkBonus + burgerBonus + pizzaBonus;
+    let total = housePrice + drinkBonus + burgerBonus + pizzaBonus;
+    return hasCFO ? Math.ceil(total * 1.5) : total;
```
#3 talks about having the text show the base price and the CFO bonus split out, can we add that in?

dan-snw replied:
Fair point, I will add this in the next set of changes when I add waitresses too since if we display CFO separately we presumably want to display waitresses separately.

egnwd commented on src/Components/RevenueBonusArea.tsx:
```diff
@@ -0,0 +1,40 @@
+    <section>
+      <Title>Revenue Bonuses</Title>
+      <div>
+        <ToggleButton
+          label="Garden"
+          active={hasGarden}
+          onClick={handleHasGarden}
+        />
+        <ToggleButton
+          label="CFO"
+          active={hasCFO}
+          onClick={handleHasCFO}
+          activeColor={ColorsToggleButton.purple}
+        />
```
Based on the screenshot we potentially need a minimum width or playing around with `display: grid` or something. min-width is definitely a lot easier here. Just looks a little weird being so thin.

dan-snw replied:
As above, will style with waitresses added since that's going to then look a bit different and grid might make more sense.

egnwd commented on src/Components/ToggleButton.tsx:
```diff
@@ -4,21 +4,35 @@ type ToggleButtonProps = {
   );
 };
 
+const ColorsToggleButton = {
+  palePink: 'rgb(246, 170, 147)',
+  purple: 'rgb(161, 122, 211)',
+};
```
Might be good to not have these in here, having another files and naming them based on the project, e.g. `const FoodChainMagnatePalette` and naming the keys similarly themed: e.g. from `palePink` to `discounters`, and `purple` to `cfo`.

dan-snw replied:
This is now fixed - I also added default colours to this palette so RGB values only appear in the palette and the CSS.

dan-snw committed:
Puts color palette in separate file

Color palette has its own file and there is now a FoodChain
MagnatePalette.

This is labelled wth the type of the component targetted rather than the
color, e.g. cfo rather than purple.

I also added defaults that are now used in the rest of the JS, so no rgb
values are to be found except in the color palette and the CSS - this
will make it easier to change colours later on.
